app/vmselect/promql/exec.go

```diff
@@ -62,6 +70,14 @@ func Exec(qt *querytracer.Tracer, ec *EvalConfig, q string, isFirstPointOnly boo
 	if err != nil {
 		return nil, err
 	}
+	if !isSubQueryComplete(e, false) {
```

Comment:
performance wise, it is better to check if flags are set before traversing the expression.

Reply:
updated)